packages/manager-react-components/src/hooks/datagrid/useIcebergV2.tsx

```diff
@@ -72,3 +84,53 @@ export function useResourcesIcebergV2<T = unknown>({
     status,
   };
 }
+
+type UseIcebergV2InfiniteQueryParams<T> = {
+  iceberg: Omit<IcebergFetchParamsV2, 'cursor'>;
+  query: Omit<
+    UseInfiniteQueryOptions,
+    'queryFn' | 'getNextPageParam' | 'initialPageParam'
+  > & {
+    select?: (data: InfiniteData<T, unknown>) => T[];
+  };
+  shouldFetchAll?: boolean;
+};
+
+export function useIcebergV2InfiniteQuery<T>({
+  iceberg: { pageSize, search, ...icebergOptions },
+  query: { queryKey, ...queryOptions },
+  shouldFetchAll = false,
```

Comment:
I think this should be a new function since it is not used the same way : on previous function you needed to fetch next page by yourselves but with this it isn't required

Reply:
I'm not sure to get the point here, this same functionality was added a few weeks back on the previous hook, it's just an helper to make it easy to fetch all the pages, of course now you will be able to handle it yourself but it's practical to have this imo